internal/controller/nats/controller.go

```diff
@@ -82,23 +86,87 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	// copy the object, so we don't modify the source object
 	nats := currentNats.DeepCopy()
 
+	// logger with nats details
+	log := r.loggerWithNATS(nats)
+
 	// check if nats is in deletion state
 	if nats.IsInDeletion() {
-		return ctrl.Result{}, nil
+		return r.handleNATSDeletion(ctx, nats, log)
 	}
 
 	// handle reconciliation
-	return ctrl.Result{}, nil
+	return r.handleNATSReconcile(ctx, nats, log)
+}
+
+// generateNatsResources renders the NATS chart with provided overrides.
+// It puts results into ReleaseInstance.
+func (r *Reconciler) generateNatsResources(nats *natsv1alpha1.NATS, instance *chart.ReleaseInstance) error {
+	// generate Nats resources from chart
+	natsResources, err := r.NATSManager.GenerateNATSResources(
+		instance,
+		manager.WithOwnerReference(*nats), // add owner references to all resources
+	)
+	if err != nil {
+		return err
+	}
+
+	// update manifests in instance
+	instance.SetRenderedManifests(*natsResources)
+	return nil
+}
+
+// initNATSInstance initializes a new NATS release instance based on NATS CR.
+func (r *Reconciler) initNATSInstance(ctx context.Context, nats *natsv1alpha1.NATS,
+	log *zap.SugaredLogger) (*chart.ReleaseInstance, error) {
+	// Init a release instance
+	instance := &chart.ReleaseInstance{
+		Name:      nats.Name,
+		Namespace: nats.Namespace,
+	}
+
+	// Check if istio is enabled in cluster
+	istioExists, err := r.kubeClient.DestinationRuleCRDExists(ctx)
```

Comment:
Does this guarantee that istio as a Kyma module is deployed and ready?
Curious how other modules are doing this check!